klippy/extras/display/menu.cfg

```diff
@@ -264,8 +276,8 @@ name: Move 10mm
 type: input
 name: Move X:{'%05.1f' % menu.input}
 input: {printer.gcode_move.gcode_position.x}
-input_min: 0
-input_max: 200
+input_min: {printer.configfile.config.stepper_x.position_min|default(0)}
+input_max: {printer.configfile.config.stepper_x.position_max|default(200)}
```

Comment:
So to make it clear I of coarse tested the lines you see there, it might not be correct for Peyton and Jinja but it is correct for the menu.cfg used by klipper

Reply:
Ok you might be right here I may need to break the search at config and look for stepper_x.position_min 

As said I have overseen that delta uses a b c instead of x y z

Reply:
Perhaps there's some unexpected difference in our configuration, but I just tried the version of the change without the default dictionary, with stepper_a substituted for stepper_z, and klipper aborted when I attempted to use the corresponding menu item from the LCD. Here's the error from the log:
```
Error evaluating 'menu __my_main __control __move_01mm __axis_z:input_min': UndefinedError: 'dict object' has no attribute 'stepper_a'
Traceback (most recent call last):
  File "/home/pi/klipper/klippy/extras/gcode_macro.py", line 61, in render
    return str(self.template.render(context))
  File "/home/pi/klippy-env/local/lib/python2.7/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/home/pi/klippy-env/local/lib/python2.7/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "<template>", line 1, in top-level template code
  File "/home/pi/klippy-env/local/lib/python2.7/site-packages/jinja2/environment.py", line 430, in getattr
    return getattr(obj, attribute)
UndefinedError: 'dict object' has no attribute 'stepper_a'
Unhandled exception during run
Traceback (most recent call last):
  File "/home/pi/klipper/klippy/klippy.py", line 188, in run
    self.reactor.run()
  File "/home/pi/klipper/klippy/reactor.py", line 269, in run
    g_next.switch()
  File "/home/pi/klipper/klippy/reactor.py", line 310, in _dispatch_loop
    timeout = self._check_timers(eventtime, busy)
  File "/home/pi/klipper/klippy/reactor.py", line 156, in _check_timers
    t.waketime = waketime = t.callback(eventtime)
  File "/home/pi/klipper/klippy/reactor.py", line 48, in invoke
    res = self.callback(eventtime)
  File "/home/pi/klipper/klippy/extras/buttons.py", line 76, in <lambda>
    (lambda e, s=self, b=nb: s.handle_button(e, b)))
  File "/home/pi/klipper/klippy/extras/buttons.py", line 82, in handle_button
    callback(eventtime, (button & mask) >> shift)
  File "/home/pi/klipper/klippy/extras/display/menu_keys.py", line 96, in click_callback
    self.callback('click', eventtime)
  File "/home/pi/klipper/klippy/extras/display/menu.py", line 1025, in key_event
    self._click_callback(eventtime, key)
  File "/home/pi/klipper/klippy/extras/display/menu.py", line 1018, in _click_callback
    self.press(event)
  File "/home/pi/klipper/klippy/extras/display/menu.py", line 931, in press
    current.run_script(event)
  File "/home/pi/klipper/klippy/extras/display/menu.py", line 231, in run_script
    _handle()
  File "/home/pi/klipper/klippy/extras/display/menu.py", line 592, in handle_script_click
    self.start_editing()
  File "/home/pi/klipper/klippy/extras/display/menu.py", line 489, in start_editing
    self._init_value()
  File "/home/pi/klipper/klippy/extras/display/menu.py", line 549, in _init_value
    self._input_min = self._eval_min(context)
  File "/home/pi/klipper/klippy/extras/display/menu.py", line 519, in _eval_min
    self._input_min_tpl.render(context)))
  File "/home/pi/klipper/klippy/extras/gcode_macro.py", line 66, in render
    raise self.gcode.error(msg)
CommandError: Error evaluating 'menu __my_main __control __move_01mm __axis_z:input_min': UndefinedError: 'dict object' has no attribute 'stepper_a'
```

Just to further clarify, the failure won't trigger until interacting with the corresponding menu item on the physical LCD, because that's when this part of the template is evaluated.

It's certainly possible I've made a mistake in my testing (I genuinely don't have much experience with Jinja). However, the Jinja documentation does state that undefined objects default to not being chainable. I don't see klipper setting the jinja2.Environment to change this, and it appears that klipper's version of Jinja (2.10.1) doesn't even support chainable undefined objects ([added in version 2.11](https://jinja.palletsprojects.com/en/2.11.x/api/#jinja2.ChainableUndefined)). So, the above error does seem entirely consistent with what the documentation states.

Reply:
As told I will fix it if needed but I will not spend any more time on it if Kevin says he won’t merge it anyhow.

This version of the limit was added by a suggestion of a other member. Either your version or my first version would work. I know that it will only have an effect while pressing it that’s the reason I checked any single menu item I have modified.